source/common/filter/tcp_proxy.cc

```diff
@@ -140,6 +140,11 @@ void TcpProxy::readDisableUpstream(bool disable) {
 
 void TcpProxy::readDisableDownstream(bool disable) {
   read_callbacks_->connection().readDisable(disable);
+  // The WsHandlerImpl class uses TCP Proxy code with a null config.
```

Comment:
The only difference between the TCP and HTTP downstream connection stats were the flow control stats. Others like 
```c++
  COUNTER(downstream_cx_rx_bytes_total)                                                            \
  GAUGE  (downstream_cx_rx_bytes_buffered)                                                         \
  COUNTER(downstream_cx_tx_bytes_total)                                                            \
  GAUGE  (downstream_cx_tx_bytes_buffered)                                                         \
  COUNTER(downstream_cx_total)                                                                     \
```
are already being recorded by HTTP. I was under the assumption that when flow control is added to HTTP, these missing stats would naturally fall in.